docs/proposals/improved-schema-for-services.md

```diff
@@ -0,0 +1,562 @@
+
+# Improved schema for operator backed services
+
+## Abstract
+This proposal is about improving the experience for a user of the operator backed services. Currently a user doesn't know exaustively which fields are available in a CR as we depend on optional metadata present in CRDs, to improve this we are considering using the metadata available from the cluster about the CRDs.
+Getting that metadata from the cluster is challenging because a normal cluster user ( plain vanilla ) doesn't have access to that metadata and so we are going to implement two workflows - one for the user that has the privileges and one that doesn't.
+This change would affect multiple service commands and the changes are described briefly below -
+- `odo catalog describe service` should include the metadata fields with description so the users can provide then when doing `odo service create`
+- `odo catalog list service` shouldn't change much other then listing CRDs per operator
+- `odo service create` should take flags and dynamically fill the CRD structs - metadata would be used for validation
+- `odo service create --input-file` would be used by adapter team or whoever wants to provide the whole CRD themselves as a file
```

Comment:
flag for this already exists and it is implemented. It is `odo service create --form-file`

Reply:
`--from-file` I assume?

Reply:
yes, that was typo